cirq/ops/eigen_gate.py

```diff
@@ -79,21 +79,20 @@ def __init__(self, *,  # Forces keyword args.
             global_shift_in_half_turns: How much to shift the eigencomponents'
                 angles by (before multiplying by the exponent).
         """
-
         self._exponent = exponent
         self._global_shift_in_half_turns = global_shift_in_half_turns
 
         # Canonicalize the exponent.
         period = self._canonical_exponent_period()
-        if period is not None and not isinstance(exponent, value.Symbol):
+        if period is not None and not isinstance(exponent,(sympy.mul.Mul,
```

Comment:
As part of this change we would actually completely delete `cirq.Symbol`.

Reply:
Yup. This is me being wary of not rocking the boat too much :D